lib/utils.js

```diff
@@ -242,10 +242,11 @@ utils.improveAjvErrors = (errors, asyncapiYAMLorJSON, initialFormat) => {
   const ast = getAST(asyncapiYAMLorJSON, initialFormat);
   return errors.map(error => {
     const defaultLocation = { jsonPointer: error.dataPath || '/' };
-
+    const additionalProperty = error.params.additionalProperty;
+    const jsonPointer = additionalProperty ? error.dataPath || `/${additionalProperty}`: error.dataPath;
```

Comment:
so you mean `error.dataPath` is not there when we talk about additional properties on the root?

Reply:
Yes. That seems to be the problem.

Reply:
> @KhudaDad414 looks good but could you please add a test that will protect us in future. Simple test where you provide some new prop on root in the document and expect an error on a given position in the document, in here https://github.com/asyncapi/parser-js/blob/master/test/parse_test.js

@derberg tests added for both scenarios. (root and non-root additional properties)